go-controller/pkg/ovnwebhook/nodeadmission.go

```diff
@@ -0,0 +1,169 @@
+package ovnwebhook
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	hotypes "github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"golang.org/x/exp/maps"
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	nodeutil "k8s.io/component-helpers/node/util"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type checkNodeAnnot func(v annotationChange, nodeName string) error
+
+// commonNodeAnnotationChecks holds annotations allowed for ovnkube-node:<nodeName> users in non-IC and IC environments
+var commonNodeAnnotationChecks = map[string]checkNodeAnnot{
+	util.OvnNodeHostAddresses:            nil,
+	util.OvnNodeL3GatewayConfig:          nil,
+	util.OvnNodeManagementPortMacAddress: nil,
+	util.OvnNodeIfAddr:                   nil,
+	util.OvnNodeChassisID: func(v annotationChange, nodeName string) error {
+		if v.action == removed {
+			return fmt.Errorf("%s cannot be removed", util.OvnNodeChassisID)
+		}
+		if v.action == changed {
+			return fmt.Errorf("%s cannot be changed once set", util.OvnNodeChassisID)
```

Comment:
`return fmt.Errorf("%s cannot be changed once set on %q", util.OvnNodeChassisID, nodeName)`

Reply:
There is no point in adding the node name here, the error in the main function already includes it:
```
return nil, fmt.Errorf("ovnkube-node on node: %q is not allowed to set %s on node %q: %v", nodeName, key, newNode.Name, err)
```